src/pages/Search.js

```diff
@@ -1,16 +1,45 @@
-import React from 'react';
-import { useParams } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { Redirect, useParams, withRouter } from 'react-router-dom';
+import SubredditForm from '../components/SubredditForm';
 import defaultSubreddit from '../helpers/constants';
 
+
 const Search = () => {
-  const { subreddit = defaultSubreddit } = useParams();
-  return (
+  const { subredditQuery } = useParams();
+  const [subreddit, setSubreddit] = useState(subredditQuery || defaultSubreddit);
+  const [route, setRoute] = useState(null);
+
+  const handleSubmit = (evt) => {
+    setRoute(`/search/${subreddit}`);
```

Comment:
Here you could use the [useHistory](https://reacttraining.com/react-router/web/api/Hooks/usehistory) hook to update the route directly.